In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('../../data/raw/stats_12-22.csv')

In [3]:
df.head()

match_id         match_home_team match_away_team  match_date  match_round  \
0     13960  Greater Western Sydney          Sydney  2012-03-24            1   
1     13960  Greater Western Sydney          Sydney  2012-03-24            1   
2     13960  Greater Western Sydney          Sydney  2012-03-24            1   
3     13960  Greater Western Sydney          Sydney  2012-03-24            1   
4     13960  Greater Western Sydney          Sydney  2012-03-24            1   

   match_home_team_goals  match_home_team_behinds  match_home_team_score  \
0                      5                        7                     37   
1                      5                        7                     37   
2                      5                        7                     37   
3                      5                        7                     37   
4                      5                        7                     37   

   match_away_team_goals  match_away_team_behinds  ...  hitout_win_percentage  \
0                     14                       16  ...                    0.0   
1                     14                       16  ...                    0.0   
2                     14                       16  ...                    0.0   
3                     14                       16  ...                    0.0   
4                     14                       16  ...                    0.0   

   intercept_marks marks_on_lead  pressure_acts rating_points ruck_contests  \
0              0.0           1.0           15.0          11.6           0.0   
1              0.0           2.0           12.0           9.8           0.0   
2              3.0           0.0           15.0          12.4           0.0   
3              1.0           0.0           23.0          15.9           0.0   
4              8.0           0.0            7.0          16.0           0.0   

  score_launches  shots_at_goal  spoils  player_position  
0            2.0            1.0     1.0                R  
1            1.0            2.0     1.0              CHF  
2            1.0            0.0     3.0               WR  
3            2.0            1.0     0.0               RR  
4            2.0            0.0     3.0              CHB  

[5 rows x 69 columns]

In [4]:
df = df.astype({'match_date': 'datetime64[ns]'})

In [5]:
df['season'] = df.match_date.dt.year

In [6]:
len(df[df.isnull().any(axis=1)])

889

In [7]:
# null values arise in these years:
df[df.isnull().any(axis=1)].season.unique()

array([2012, 2013, 2014, 2015, 2016, 2020, 2021], dtype=int32)

# Investigating 2012-2016 values

It seems for whatever odd reason, there are 2 or 3 matches in each season where the advanced statistics were not recorded
I have searched the official AFL website as well as other common sources such as afl tables and footywire, but they all return n/a.

In [8]:
df_2012 = df.query('season == 2012')
df_2012_na = df_2012[df_2012.isna().any(axis=1)]

In [9]:
df_2012.isna().sum().sort_values(ascending=False)[:25]

contest_off_wins           88
ground_ball_gets           88
score_involvements         88
metres_gained              88
turnovers                  88
intercepts                 88
tackles_inside_fifty       88
contest_def_losses         88
contest_def_one_on_ones    88
contest_off_one_on_ones    88
def_half_pressure_acts     88
effective_kicks            88
f50_ground_ball_gets       88
hitouts_to_advantage       88
hitout_win_percentage      88
intercept_marks            88
marks_on_lead              88
pressure_acts              88
ruck_contests              88
score_launches             88
shots_at_goal              88
spoils                     88
afl_fantasy_score           0
bounces                     0
one_percenters              0
dtype: int64

In [10]:
# the n/a values arise in 2 matches: Adelaide, GWS in round 4 and GC, Carlton round 22
# it seems the values were not recorded as they are not present on the official AFL website, or any other data keeping site that I have seen

df_2012_na

match_id match_home_team         match_away_team match_date  \
1320     13990        Adelaide  Greater Western Sydney 2012-04-21   
1321     13990        Adelaide  Greater Western Sydney 2012-04-21   
1322     13990        Adelaide  Greater Western Sydney 2012-04-21   
1323     13990        Adelaide  Greater Western Sydney 2012-04-21   
1324     13990        Adelaide  Greater Western Sydney 2012-04-21   
...        ...             ...                     ...        ...   
8135     14144      Gold Coast                 Carlton 2012-08-25   
8136     14144      Gold Coast                 Carlton 2012-08-25   
8137     14144      Gold Coast                 Carlton 2012-08-25   
8138     14144      Gold Coast                 Carlton 2012-08-25   
8139     14144      Gold Coast                 Carlton 2012-08-25   

      match_round  match_home_team_goals  match_home_team_behinds  \
1320            4                     13                       18   
1321            4                     13                       18   
1322            4                     13                       18   
1323            4                     13                       18   
1324            4                     13                       18   
...           ...                    ...                      ...   
8135           22                     15                        8   
8136           22                     15                        8   
8137           22                     15                        8   
8138           22                     15                        8   
8139           22                     15                        8   

      match_home_team_score  match_away_team_goals  match_away_team_behinds  \
1320                     96                      7                        8   
1321                     96                      7                        8   
1322                     96                      7                        8   
1323                     96                      7                        8   
1324                     96                      7                        8   
...                     ...                    ...                      ...   
8135                     98                     11                       20   
8136                     98                     11                       20   
8137                     98                     11                       20   
8138                     98                     11                       20   
8139                     98                     11                       20   

      ...  intercept_marks  marks_on_lead pressure_acts  rating_points  \
1320  ...              NaN            NaN           NaN           10.2   
1321  ...              NaN            NaN           NaN           15.0   
1322  ...              NaN            NaN           NaN            4.2   
1323  ...              NaN            NaN           NaN            8.0   
1324  ...              NaN            NaN           NaN            7.7   
...   ...              ...            ...           ...            ...   
8135  ...              NaN            NaN           NaN           12.6   
8136  ...              NaN            NaN           NaN            0.0   
8137  ...              NaN            NaN           NaN           12.1   
8138  ...              NaN            NaN           NaN           -2.5   
8139  ...              NaN            NaN           NaN            0.3   

     ruck_contests score_launches shots_at_goal  spoils  player_position  \
1320           NaN            NaN           NaN     NaN             HFFL   
1321           NaN            NaN           NaN     NaN             HBFL   
1322           NaN            NaN           NaN     NaN              CHB   
1323           NaN            NaN           NaN     NaN              BPR   
1324           NaN            NaN           NaN     NaN               RK   
...            ...            ...           ...     ..

# 2013

In [11]:
df_2013 = df.query('season == 2013')
df_2013_na = df_2013[df_2013.isna().any(axis=1)]

In [12]:
df_2013.isna().sum().sort_values(ascending=False)[:25]

contest_off_wins           88
ground_ball_gets           88
score_involvements         88
metres_gained              88
turnovers                  88
intercepts                 88
tackles_inside_fifty       88
contest_def_losses         88
contest_def_one_on_ones    88
contest_off_one_on_ones    88
def_half_pressure_acts     88
effective_kicks            88
f50_ground_ball_gets       88
hitouts_to_advantage       88
hitout_win_percentage      88
intercept_marks            88
marks_on_lead              88
pressure_acts              88
ruck_contests              88
score_launches             88
shots_at_goal              88
spoils                     88
afl_fantasy_score           0
bounces                     0
one_percenters              0
dtype: int64

In [13]:
df_2013_na

match_id match_home_team  match_away_team match_date  match_round  \
9504      14185       Fremantle         Essendon 2013-04-12            3   
9505      14185       Fremantle         Essendon 2013-04-12            3   
9506      14185       Fremantle         Essendon 2013-04-12            3   
9507      14185       Fremantle         Essendon 2013-04-12            3   
9508      14185       Fremantle         Essendon 2013-04-12            3   
...         ...             ...              ...        ...          ...   
14515     14298  Brisbane Lions  North Melbourne 2013-07-13           16   
14516     14298  Brisbane Lions  North Melbourne 2013-07-13           16   
14517     14298  Brisbane Lions  North Melbourne 2013-07-13           16   
14518     14298  Brisbane Lions  North Melbourne 2013-07-13           16   
14519     14298  Brisbane Lions  North Melbourne 2013-07-13           16   

       match_home_team_goals  match_home_team_behinds  match_home_team_score  \
9504                       9                       14                     68   
9505                       9                       14                     68   
9506                       9                       14                     68   
9507                       9                       14                     68   
9508                       9                       14                     68   
...                      ...                      ...                    ...   
14515                     18                        9                    117   
14516                     18                        9                    117   
14517                     18                        9                    117   
14518                     18                        9                    117   
14519                     18                        9                    117   

       match_away_team_goals  match_away_team_behinds  ...  intercept_marks  \
9504                      10                       12  ...              NaN   
9505                      10                       12  ...              NaN   
9506                      10                       12  ...              NaN   
9507                      10                       12  ...              NaN   
9508                      10                       12  ...              NaN   
...                      ...                      ...  ...              ...   
14515                     17                        3  ...              NaN   
14516                     17                        3  ...              NaN   
14517                     17                        3  ...              NaN   
14518                     17                        3  ...              NaN   
14519                     17                        3  ...              NaN   

       marks_on_lead pressure_acts  rating_points ruck_contests  \
9504             NaN           NaN            8.0           NaN   
9505             NaN           NaN           -0.2           NaN   
9506             NaN           NaN            0.0           NaN   
9507             NaN           NaN           10.2           NaN   
9508             NaN           NaN            1.5           NaN   
...              ...           ...            ...           ...   
14515            NaN           NaN            9.6           NaN   
14516            NaN           NaN           10.0           NaN   
14517            NaN           NaN            6.2           NaN   
14518            NaN           NaN           19.0           NaN   
14519            NaN           NaN            9.5           NaN   

      score_launches shots_at_goal  spoils  player_position  season  
9504             NaN           NaN     NaN              BPR    2013  
9505             NaN           NaN     NaN              CHF    2013  
9506             NaN           NaN     NaN              CHB    2013  
9507             NaN           NaN     NaN             HBFL    2013  
9508             NaN         

In [14]:
df_2014 = df.query('season == 2014')
df_2014_na = df_2014[df_2014.isna().any(axis=1)]

In [15]:
df_2014.isna().sum().sort_values(ascending=False)[:25]

contest_off_wins           88
ground_ball_gets           88
score_involvements         88
metres_gained              88
turnovers                  88
intercepts                 88
tackles_inside_fifty       88
contest_def_losses         88
contest_def_one_on_ones    88
contest_off_one_on_ones    88
def_half_pressure_acts     88
effective_kicks            88
f50_ground_ball_gets       88
hitouts_to_advantage       88
hitout_win_percentage      88
intercept_marks            88
marks_on_lead              88
pressure_acts              88
ruck_contests              88
score_launches             88
shots_at_goal              88
spoils                     88
afl_fantasy_score           0
bounces                     0
one_percenters              0
dtype: int64

In [16]:
df_2014_na

match_id match_home_team match_away_team match_date  match_round  \
20812     14451     Collingwood      West Coast 2014-05-24           10   
20813     14451     Collingwood      West Coast 2014-05-24           10   
20814     14451     Collingwood      West Coast 2014-05-24           10   
20815     14451     Collingwood      West Coast 2014-05-24           10   
20816     14451     Collingwood      West Coast 2014-05-24           10   
...         ...             ...             ...        ...          ...   
25207     14550     Collingwood  Brisbane Lions 2014-08-16           21   
25208     14550     Collingwood  Brisbane Lions 2014-08-16           21   
25209     14550     Collingwood  Brisbane Lions 2014-08-16           21   
25210     14550     Collingwood  Brisbane Lions 2014-08-16           21   
25211     14550     Collingwood  Brisbane Lions 2014-08-16           21   

       match_home_team_goals  match_home_team_behinds  match_home_team_score  \
20812                     17                        7                    109   
20813                     17                        7                    109   
20814                     17                        7                    109   
20815                     17                        7                    109   
20816                     17                        7                    109   
...                      ...                      ...                    ...   
25207                      8                        8                     56   
25208                      8                        8                     56   
25209                      8                        8                     56   
25210                      8                        8                     56   
25211                      8                        8                     56   

       match_away_team_goals  match_away_team_behinds  ...  intercept_marks  \
20812                     15                       11  ...              NaN   
20813                     15                       11  ...              NaN   
20814                     15                       11  ...              NaN   
20815                     15                       11  ...              NaN   
20816                     15                       11  ...              NaN   
...                      ...                      ...  ...              ...   
25207                     18                       15  ...              NaN   
25208                     18                       15  ...              NaN   
25209                     18                       15  ...              NaN   
25210                     18                       15  ...              NaN   
25211                     18                       15  ...              NaN   

       marks_on_lead pressure_acts  rating_points ruck_contests  \
20812            NaN           NaN           20.0           NaN   
20813            NaN           NaN           20.5           NaN   
20814            NaN           NaN            9.5           NaN   
20815            NaN           NaN            4.0           NaN   
20816            NaN           NaN           13.7           NaN   
...              ...           ...            ...           ...   
25207            NaN           NaN            4.8           NaN   
25208            NaN           NaN            8.8           NaN   
25209            NaN           NaN           10.8           NaN   
25210            NaN           NaN           12.8           NaN   
25211            NaN           NaN            9.3           NaN   

      score_launches shots_at_goal  spoils  player_position  season  
20812            NaN           NaN     NaN               FB    2014  
20813            NaN           NaN     NaN              INT    2014  
20814            NaN           NaN     NaN              INT    2014  
20815            NaN           NaN     NaN              FPR    2014  
20816            NaN           NaN     Na

In [17]:
df_2015 = df.query('season == 2015')
df_2015_na = df_2015[df_2015.isna().any(axis=1)]

In [18]:
df_2015.isna().sum().sort_values(ascending=False)[:25]

contest_off_wins             88
intercept_marks              88
contest_def_losses           88
contest_def_one_on_ones      88
contest_off_one_on_ones      88
def_half_pressure_acts       88
effective_kicks              88
f50_ground_ball_gets         88
ground_ball_gets             88
hitout_win_percentage        88
hitouts_to_advantage         88
marks_on_lead                88
pressure_acts                88
ruck_contests                88
score_launches               88
spoils                       88
score_involvements            0
bounces                       0
goal_assists                  0
one_percenters                0
marks_inside_fifty            0
time_on_ground_percentage     0
afl_fantasy_score             0
centre_clearances             0
stoppage_clearances           0
dtype: int64

In [19]:
df_2015_na

match_id match_home_team match_away_team match_date  match_round  \
26532     14590      West Coast         Carlton 2015-04-10            2   
26533     14590      West Coast         Carlton 2015-04-10            2   
26534     14590      West Coast         Carlton 2015-04-10            2   
26535     14590      West Coast         Carlton 2015-04-10            2   
26536     14590      West Coast         Carlton 2015-04-10            2   
...         ...             ...             ...        ...          ...   
31323     14698        Richmond         Carlton 2015-07-10           15   
31324     14698        Richmond         Carlton 2015-07-10           15   
31325     14698        Richmond         Carlton 2015-07-10           15   
31326     14698        Richmond         Carlton 2015-07-10           15   
31327     14698        Richmond         Carlton 2015-07-10           15   

       match_home_team_goals  match_home_team_behinds  match_home_team_score  \
26532                     20                       11                    131   
26533                     20                       11                    131   
26534                     20                       11                    131   
26535                     20                       11                    131   
26536                     20                       11                    131   
...                      ...                      ...                    ...   
31323                     10                       11                     71   
31324                     10                       11                     71   
31325                     10                       11                     71   
31326                     10                       11                     71   
31327                     10                       11                     71   

       match_away_team_goals  match_away_team_behinds  ...  intercept_marks  \
26532                      9                        8  ...              NaN   
26533                      9                        8  ...              NaN   
26534                      9                        8  ...              NaN   
26535                      9                        8  ...              NaN   
26536                      9                        8  ...              NaN   
...                      ...                      ...  ...              ...   
31323                      5                       11  ...              NaN   
31324                      5                       11  ...              NaN   
31325                      5                       11  ...              NaN   
31326                      5                       11  ...              NaN   
31327                      5                       11  ...              NaN   

       marks_on_lead pressure_acts  rating_points ruck_contests  \
26532            NaN           NaN            8.3           NaN   
26533            NaN           NaN           17.7           NaN   
26534            NaN           NaN            8.4           NaN   
26535            NaN           NaN            7.8           NaN   
26536            NaN           NaN           23.1           NaN   
...              ...           ...            ...           ...   
31323            NaN           NaN            2.5           NaN   
31324            NaN           NaN            5.7           NaN   
31325            NaN           NaN           11.3           NaN   
31326            NaN           NaN            5.4           NaN   
31327            NaN           NaN            3.1           NaN   

      score_launches shots_at_goal  spoils  player_position  season  
26532            NaN           0.0     NaN                R    2015  
26533            NaN           0.0     NaN              INT    2015  
26534            NaN           0.0     NaN                C    2015  
26535            NaN           0.0     NaN               RK    2015  
26536            NaN           6.0     Na

In [20]:
df_2016 = df.query('season == 2016')
df_2016_na = df_2016[df_2016.isna().any(axis=1)]

In [21]:
df_2016.isna().sum().sort_values(ascending=False)[:25]

contest_off_wins             132
intercept_marks              132
contest_def_losses           132
contest_def_one_on_ones      132
contest_off_one_on_ones      132
def_half_pressure_acts       132
effective_kicks              132
f50_ground_ball_gets         132
ground_ball_gets             132
hitout_win_percentage        132
hitouts_to_advantage         132
marks_on_lead                132
pressure_acts                132
ruck_contests                132
score_launches               132
spoils                       132
score_involvements             0
bounces                        0
goal_assists                   0
one_percenters                 0
marks_inside_fifty             0
time_on_ground_percentage      0
afl_fantasy_score              0
centre_clearances              0
stoppage_clearances            0
dtype: int64

In [22]:
df_2016_na.iloc[[0, 60, 130]]

match_id match_home_team match_away_team match_date  match_round  \
34980     14791          Sydney     Collingwood 2016-03-26            1   
37592     14850        Adelaide         Geelong 2016-05-13            8   
40522     14916      Gold Coast  Brisbane Lions 2016-07-09           16   

       match_home_team_goals  match_home_team_behinds  match_home_team_score  \
34980                     18                       25                    133   
37592                     11                        6                     72   
40522                     22                        7                    139   

       match_away_team_goals  match_away_team_behinds  ...  intercept_marks  \
34980                      7                       11  ...              NaN   
37592                     13                       20  ...              NaN   
40522                     17                       11  ...              NaN   

       marks_on_lead pressure_acts  rating_points ruck_contests  \
34980            NaN           NaN            0.0           NaN   
37592            NaN           NaN           18.0           NaN   
40522            NaN           NaN           10.1           NaN   

      score_launches shots_at_goal  spoils  player_position  season  
34980            NaN           0.0     NaN              INT    2016  
37592            NaN           1.0     NaN              INT    2016  
40522            NaN           0.0     NaN              FPR    2016  

[3 rows x 70 columns]

# Investigating 2020-2021 values

In [23]:
df_2020 = df.query('season == 2020')
df_2020_na = df_2020[df_2020.isna().any(axis=1)]

In [24]:
df_2020.isna().sum().sort_values(ascending=False)[:2]

afl_fantasy_score    395
match_id               0
dtype: int64

In [25]:
# afl fantasy scores were not recorded in dataset in all 9 round 1 matches
len(df_2020_na.match_id.unique())

9

In [26]:
idx_to_change = df_2020_na.index

In [27]:
# calculate fantasy score via formula on afl website
# https://www.fantasy.afl.com.au/classic/help/game-guidelines

df.loc[idx_to_change, 'afl_fantasy_score'] = 3*df['kicks'] + 2*df['handballs'] \
                                            + 3*df['marks'] +4*df['tackles'] \
                                            + 1*df['free_kicks_for'] \
                                            - 3*df['free_kicks_against'] \
                                            + df['hitouts'] \
                                            + 6*df['goals'] \
                                            + 1*df['behinds']

In [28]:
# visible values agree with AFL website
df.loc[idx_to_change][['player_first_name', 'player_last_name', 'afl_fantasy_score']]

player_first_name player_last_name  afl_fantasy_score
69652              Kade          Simpson               72.0
69653              Marc           Murphy               78.0
69654             Shane          Edwards               52.0
69655              Jack         Riewoldt               50.0
69656           Matthew          Kreuzer               16.0
...                 ...              ...                ...
70043              Jack      Petruccelle               36.0
70044            Jarrod          Brander               61.0
70045               Jay         Lockhart               42.0
70046              Toby          Bedford               26.0
70047           Kysaiah          Pickett               44.0

[395 rows x 3 columns]

# Editing 2021 values

In [29]:
df_2021 = df.query('season == 2021')

In [30]:
df_2021_na = df_2021[df_2021.isna().any(axis=1)]

In [31]:
# only 10 player's rating points are missing
df_2021.isna().sum().sort_values(ascending=False)[:2]

rating_points    10
match_id          0
dtype: int64

In [32]:
col = [
    'match_home_team', 'match_away_team', 'match_round',
    'player_first_name', 'player_last_name', 'rating_points', 'disposals'
]

In [33]:
# create table to easily find missing values on AFL website
df_2021_na[col]

match_home_team         match_away_team  match_round  \
78198            Sydney  Greater Western Sydney            5   
78644         Melbourne                Richmond            6   
79874        Gold Coast          Brisbane Lions            9   
79993          Essendon               Fremantle            9   
80297  Western Bulldogs                St Kilda           10   
82092        West Coast        Western Bulldogs           15   
82155           Carlton                Adelaide           15   
83861          St Kilda                 Carlton           20   
84619        West Coast               Melbourne           21   
85073         Fremantle              West Coast           22   

      player_first_name player_last_name  rating_points  disposals  
78198            Oliver          Florent            NaN         15  
78644              Jake         Melksham            NaN         11  
79874              Izak          Rankine            NaN          5  
79993             Darcy           Parish            NaN         39  
80297             Shaun         McKernan            NaN         11  
82092              Luke            Shuey            NaN         20  
82155           Matthew          Kennedy            NaN         24  
83861            Thomas       Williamson            NaN         13  
84619           Josh J.          Kennedy            NaN         10  
85073              Luke            Foley            NaN         13

In [34]:
# fortunately, only 10 errors means it is easy to source by hand
rating_points_in_order = [
    7.3,
    0.5,
    2.8,
    19.8,
    3.3,
    10.6,
    4.8,
    8.5,
    14.3,
    5.5
]

In [35]:
idx_to_change = df_2021_na.index

In [36]:
df.loc[idx_to_change, 'rating_points'] = rating_points_in_order

In [37]:
# check values entered as intended
df.loc[idx_to_change, ['player_last_name', 'rating_points']]

player_last_name  rating_points
78198          Florent            7.3
78644         Melksham            0.5
79874          Rankine            2.8
79993           Parish           19.8
80297         McKernan            3.3
82092            Shuey           10.6
82155          Kennedy            4.8
83861       Williamson            8.5
84619          Kennedy           14.3
85073            Foley            5.5

# Final Comments

To begin, there was 889 total rows of n/a values
After investigating the reason for the missing values, only 484 remain.

These 484 correspond with 11 games * 44 players where individual statistics were simply 
not recorded during the play. This means a total of 33 brownlow polling performances will not 
be learned by the model.

Fortunately the missing performances occured in 2012-1016 and therefore do not impact
the years 2021 and 2022, which will be used for testing the model.

In [38]:
df[df.isna().any(axis=1)]

match_id match_home_team         match_away_team match_date  \
1320      13990        Adelaide  Greater Western Sydney 2012-04-21   
1321      13990        Adelaide  Greater Western Sydney 2012-04-21   
1322      13990        Adelaide  Greater Western Sydney 2012-04-21   
1323      13990        Adelaide  Greater Western Sydney 2012-04-21   
1324      13990        Adelaide  Greater Western Sydney 2012-04-21   
...         ...             ...                     ...        ...   
40519     14916      Gold Coast          Brisbane Lions 2016-07-09   
40520     14916      Gold Coast          Brisbane Lions 2016-07-09   
40521     14916      Gold Coast          Brisbane Lions 2016-07-09   
40522     14916      Gold Coast          Brisbane Lions 2016-07-09   
40523     14916      Gold Coast          Brisbane Lions 2016-07-09   

       match_round  match_home_team_goals  match_home_team_behinds  \
1320             4                     13                       18   
1321             4                     13                       18   
1322             4                     13                       18   
1323             4                     13                       18   
1324             4                     13                       18   
...            ...                    ...                      ...   
40519           16                     22                        7   
40520           16                     22                        7   
40521           16                     22                        7   
40522           16                     22                        7   
40523           16                     22                        7   

       match_home_team_score  match_away_team_goals  match_away_team_behinds  \
1320                      96                      7                        8   
1321                      96                      7                        8   
1322                      96                      7                        8   
1323                      96                      7                        8   
1324                      96                      7                        8   
...                      ...                    ...                      ...   
40519                    139                     17                       11   
40520                    139                     17                       11   
40521                    139                     17                       11   
40522                    139                     17                       11   
40523                    139                     17                       11   

       ...  intercept_marks  marks_on_lead pressure_acts  rating_points  \
1320   ...              NaN            NaN           NaN           10.2   
1321   ...              NaN            NaN           NaN           15.0   
1322   ...              NaN            NaN           NaN            4.2   
1323   ...              NaN            NaN           NaN            8.0   
1324   ...              NaN            NaN           NaN            7.7   
...    ...              ...            ...           ...            ...   
40519  ...              NaN            NaN           NaN           14.4   
40520  ...              NaN            NaN           NaN            1.9   
40521  ...              NaN            NaN           NaN           12.5   
40522  ...              NaN            NaN           NaN           10.1   
40523  ...              NaN            NaN           NaN           13.1   

      ruck_contests score_launches shots_at_goal  spoils  player_position  \
1320            NaN            NaN           NaN     NaN             HFFL   
1321            NaN            NaN           NaN     NaN             HBFL   
1322            NaN            NaN           NaN     NaN              CHB   
1323            NaN            NaN           NaN     NaN              BPR   
1324            NaN            NaN           NaN     NaN               RK   
.

In [39]:
df.dropna(inplace=True)

In [40]:
df.to_csv('../../data/raw/stats_12-22_no_na.csv', index=False)